pybamm/models/full_battery_models/lithium_ion/base_lithium_ion_model.py

```diff
@@ -232,14 +236,40 @@ def set_sei_submodel(self):
         else:
             reaction_loc = "full electrode"
 
-        if self.options["SEI"] == "none":
-            self.submodels["sei"] = pybamm.sei.NoSEI(self.param, self.options)
-        elif self.options["SEI"] == "constant":
-            self.submodels["sei"] = pybamm.sei.ConstantSEI(self.param, self.options)
+        # Do not set "sei on cracks" submodel for half-cells
+        if reaction_loc == "interface":
+            if self.options["SEI"] == "none":
+                self.submodels["sei"] = pybamm.sei.NoSEI(self.param, self.options)
+            elif self.options["SEI"] == "constant":
+                self.submodels["sei"] = pybamm.sei.ConstantSEI(self.param, self.options)
+            else:
+                self.submodels["sei"] = pybamm.sei.SEIGrowth(
+                    self.param, reaction_loc, self.options, cracks=False
+                )
+        # For full cells, "sei on cracks" submodel must be set, even if it is zero
         else:
-            self.submodels["sei"] = pybamm.sei.SEIGrowth(
-                self.param, reaction_loc, self.options
-            )
+            if self.options["SEI"] == "none":
+                self.submodels["sei"] = pybamm.sei.NoSEI(self.param, self.options)
+                self.submodels["sei on cracks"] = pybamm.sei.NoSEI(
+                    self.param, self.options, cracks=True
+                )
+            elif self.options["SEI"] == "constant":
+                self.submodels["sei"] = pybamm.sei.ConstantSEI(self.param, self.options)
+                self.submodels["sei on cracks"] = pybamm.sei.NoSEI(
+                    self.param, self.options, cracks=True
+                )
+            else:
+                self.submodels["sei"] = pybamm.sei.SEIGrowth(
+                    self.param, reaction_loc, self.options, cracks=False
+                )
+                if self.options["SEI on cracks"] == "true":
+                    self.submodels["sei on cracks"] = pybamm.sei.SEIGrowth(
+                        self.param, reaction_loc, self.options, cracks=True
+                    )
+                else:
+                    self.submodels["sei on cracks"] = pybamm.sei.NoSEI(
+                        self.param, self.options, cracks=True
+                    )
```

Comment:
I think this does the same thing in fewer lines
```python
        if self.options["SEI"] == "none":
            self.submodels["sei"] = pybamm.sei.NoSEI(self.param, self.options)
        elif self.options["SEI"] == "constant":
            self.submodels["sei"] = pybamm.sei.ConstantSEI(self.param, self.options)
        else:
            self.submodels["sei"] = pybamm.sei.SEIGrowth(
                self.param, reaction_loc, self.options, cracks=False
            )
        # Do not set "sei on cracks" submodel for half-cells
        # For full cells, "sei on cracks" submodel must be set, even if it is zero
        if reaction_loc != "interface":
            if self.options["SEI"] in ["none", "constant"] or self.options["SEI on cracks"] == "false":
                self.submodels["sei on cracks"] = pybamm.sei.NoSEI(
                    self.param, self.options, cracks=True
                )
            else:
                self.submodels["sei on cracks"] = pybamm.sei.SEIGrowth(
                    self.param, reaction_loc, self.options, cracks=True
                )
```

Reply:
Fixed